src/lib/spsp.js

```diff
@@ -0,0 +1,152 @@
+  })
+}
+
+const quoteDestination = (client, identifier, amount) => {
+  return co(function * () {
+    if (!amount) throw new Error('missing amount')
+    return yield _quote(client, identifier, undefined, amount)
+  })
+}
+
+const sendPayment = (client, payment) => {
+  return co(function * () {
+    if (!client) throw new Error('missing client')
+    if (!payment) throw new Error('missing payment object')
+    if (!payment.destinationAccount) throw new Error('missing payment.destinationAccount')
+    if (!payment.destinationAmount) throw new Error('missing payment.destinationAmount')
+    if (!payment.sourceAmount) throw new Error('missing payment.sourceAmount')
+    if (!payment.connectorAccount) throw new Error('missing payment.connectorAccount')
+    if (!payment.receiver) throw new Error('missing payment.receiver')
+
+    const request = yield _setup(payment.receiver, payment.destinationAmount)
+    const sender = Sender.createSender({ client })
+    const fulfillment = yield sender.payRequest({
+      sourceAmount: String(payment.sourceAmount),
+      connectorAccount: payment.connectorAccount,
+      destinationAmount: String(payment.destinationAmount),
+      destinationAccount: request.address,
+      destinationMemo: {
+        data: request.data,
+        expires_at: request.expires_at
+      },
+      expiresAt: request.expires_at,
+      executionCondition: request.condition
+    })
+
+    return { fulfillment }
+  })
+}
+
+/** Contains SPSP methods */
+class Client {
+  /**
+    * Create an SPSP client.
+    * @param {Object} opts plugin options
+    * @param {Function} opts._plugin (optional) plugin constructor. Defaults to PluginBells
+    */
+  constructor (opts) {
+    this.client = new IlpCore.Client(Object.assign({
+      _plugin: PluginBells
+    }, opts))
+
+    /**
+      * Get payment params via SPSP request, based on source amount
+      * @param {String} identifier webfinger identifier of receiver
+      * @param {String} sourceAmount Amount that you will send
+      * @returns {Promise.<PaymentParams>} Resolves with the parameters that can be passed to sendPayment
+      */
+    this.quoteSource = quoteSource.bind(null, this.client)
+
+    /**
+      * Get payment params via SPSP request, based on destination amount
+      * @param {String} identifier webfinger identifier of receiver
+      * @param {String} destinationAmount Amount that the receiver will get
+      * @returns {Promise.<PaymentParams>} Resolves with the parameters that can be passed to sendPayment
+      */
+    this.quoteDestination = quoteDestination.bind(null, this.client)
+
+    /**
+      * Sends a payment using the PaymentParams
+      * @param {PaymentParams} payment params, returned by quoteSource or quoteDestination
+      * @returns {Promise.<PaymentResult>} Returns payment result
+      */
+    this.sendPayment = sendPayment.bind(null, this.client)
+  }
+}
+
+module.exports = {
+  Client,
+  quoteSource,
+  quoteDestination,
+  sendPayment
```

Comment:
Even though it's just an HTTP GET I think we should export a query function here. Maybe it would validate the schema or give you some error if that isn't a receiver